worker/operations.py

```diff
@@ -463,6 +463,9 @@ def data_card(*args, **kwargs):
 
     if "metadata" not in dataset_json:
```

Comment:
@YohannParis I think the specific issue you're seeing is actually that `metadata` _IS_ in `dataset_json` but is `None`. So you'd change line 464 to something like:

`if dataset_json.get("metadata") is None:`

Since if the key doesn't exist `get` returns `None` this should do the trick.

Below you could do the same thing for `data_card`